### PR TITLE
MONGOCRYPT-613 Rename DerivedFromDataTokenAndContentionFactor Token

### PR DESCRIPTION
**Pull request by sgolemon**
This is about 95%
 
 ```bash
 grep -r DerivedFromDataTokenAndCounter . | \
     sort | cut -d : -f 1 | uniq | \
     xargs -L 1 sed -i -e 's/DerivedFromDataTokenAndCounter/DerivedFromDataTokenAndContentionFactor/g'
 ```
 
The other 5% is some surgery in `mongocrypt-marking.c` to update some variable names.  The good news is despite having names like `counter` we still managed to pass `contentionFactor`.

### CODE REVIEW

File: src/mongocrypt-marking.c [hunk 1]
```diff
@@ -235,24 +237,27 @@ void _mongocrypt_marking_cleanup(_mongocrypt_marking_t *marking) {
...
             /* FindEqualityPayload uses *fromDataToken */                                                              \
             _mongocrypt_buffer_copy_to(mc_##Name##DerivedFromDataToken_get(fromDataToken), out);                       \
             mc_##Name##DerivedFromDataToken_destroy(fromDataToken);                                                    \
             return true;                                                                                               \
         }                                                                                                              \
                                                                                                                        \
-        BSON_ASSERT(counter >= 0);                                                                                     \
+        BSON_ASSERT(contentionFactor >= 0);                                                                            \
         /* InsertUpdatePayload continues through *fromDataTokenAndCounter */                                           \
```
Comment (erwee) [inline src/mongocrypt-marking.c:248]:
```suggestion
        /* InsertUpdatePayload continues through *fromDataTokenAndContentionFactor */                                           \
```

File: src/mongocrypt-marking.c [hunk 2]
```diff
@@ -235,24 +237,27 @@ void _mongocrypt_marking_cleanup(_mongocrypt_marking_t *marking) {
...
-        mc_##Name##DerivedFromDataTokenAndCounter_t *fromTokenAndCounter =                                             \
-            mc_##Name##DerivedFromDataTokenAndCounter_new(crypto, fromDataToken, (uint64_t)counter, status);           \
+        mc_##Name##DerivedFromDataTokenAndContentionFactor_t *fromTokenAndCounter =                                    \
```
Comment (erwee) [inline src/mongocrypt-marking.c:249]:
```suggestion
        mc_##Name##DerivedFromDataTokenAndContentionFactor_t *fromTokenAndContentionFactor =                                    \
```

**Review by erwee — CHANGES_REQUESTED** [review]
A couple of areas missed.

File: src/mongocrypt-marking.c (hunk 1 above)
Comment (sgolemon) [reply src/mongocrypt-marking.c]: Right. Plus all the local vars in this block. Good catch. Done and done.

File: src/mongocrypt-marking.c (hunk 2 above)
Comment (sgolemon) [reply src/mongocrypt-marking.c]: Done

**Review by erwee — APPROVED** [review]

**Comment by sgolemon** [comment]
> Thank you for the rename.
> 
> Suggest updating other occurrences of "andCounter" and "counter". Here is a branch with suggested additional commits: https://github.com/mongodb/libmongocrypt/compare/master...kevinAlbs:libmongocrypt:MC-613?expand=1
> 

Yep, go big or go home.

Done.

**Pull request merged**